### PR TITLE
New: `no-tabs` rule (fixes #6079)

### LINKED ISSUE #6079: Rule Proposal: no-tabs
From [disallowTabs].

This rule will disallow a use of tabs.
I guess this rule is really similar to [no-irregular-whitespace] rule.
So the option of [no-irregular-whitespace] may be better. Thoughts?

```js
{
    "no-tabs": "error"
}
```

[disallowTabs]: http://jscs.info/rule/disallowTabs
[no-irregular-whitespace]: http://eslint.org/docs/rules/no-irregular-whitespace

### PR DESCRIPTION
**Pull request by gyandeeps**
**What issue does this pull request address?**
New rule named `no-tabs`

**What changes did you make? (Give an overview)**
created a new rule.

**Is there anything you'd like reviewers to focus on?**
nothing specific

Need suggestions on:

1. Error message
2. Should we activate this rule for eslint project

### CODE REVIEW

**Comment by mysticatea** [comment]
1. `"Unexpected tab character."`?
2. Adding to our `.eslintrc` sounds good to me.

File: docs/rules/no-tabs.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Disallow tabs in file (no-tabs)
+
+Some people prefer whitespaces instead of tab characters in a file. So they would want to disallow tab anywhere inside a file including comments.
```
Comment (nzakas) [inline docs/rules/no-tabs.md]: Tabs are also whitespace. Better to say that some style guides don't allow the use of tab characters anywhere.

File: docs/rules/no-tabs.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+/**
+* its a test function
+*/
+function test(){}
+
+var x = 1; // test
+```
+
+## When Not To Use It
+
```
Comment (nzakas) [inline docs/rules/no-tabs.md]: Please add Compatibility section with JSCS equivalent

File: lib/rules/no-tabs.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/**
+ * @fileoverview Rule to check for tabs inside a file
+ * @author Gyandeep Singh
+ */
+
+"use strict";
```
Comment (nzakas) [inline lib/rules/no-tabs.js]: Can you add in the section separators?

**Comment by nzakas** [comment]
I'm :+1: fit adding to our config. Dogfood all the things!

File: lib/rules/no-tabs.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+                    if (match) {
+                        context.report(
+                            node,
+                            {
+                                line: index + 1,
+                                column: match.index + 1
+                            },
+                            "Line has tabs."
+                        );
```
Comment (nzakas) [inline lib/rules/no-tabs.js]: "Unexpected tab character."?

**Comment by gyandeeps** [comment]
All the changes have been made.

File: lib/rules/no-tabs.js (hunk 3 above)
Comment (gyandeeps) [inline lib/rules/no-tabs.js]: I dint understand what you mean here

Comment (nzakas) [inline lib/rules/no-tabs.js]: http://eslint.org/docs/developer-guide/code-conventions#file-format

Just missing the "Public Interface" separator.

Comment (gyandeeps) [inline lib/rules/no-tabs.js]: ah i see. I was focusing on use script, as i thought it needs to be moved some where.

Comment (gyandeeps) [inline lib/rules/no-tabs.js]: done.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**